packages/apollo-gateway/src/index.ts

```diff
@@ -105,10 +105,12 @@ export type Experimental_DidResolveQueryPlanCallback = ({
   queryPlan,
   serviceMap,
   operationContext,
+  requestContext,
 }: {
   readonly queryPlan: QueryPlan;
   readonly serviceMap: ServiceMap;
   readonly operationContext: OperationContext;
+  readonly requestContext?: GraphQLRequestContextExecutionDidStart<Record<string, any>>;
```

Comment:
I think the `?:` is needed for backwards compatibility? Also I'm not sure if `Record<string, any>` is right.

Reply:
As long as `this.experimental_didResolveQueryPlan` is called with this object in all cases (it is, in its one usage below), then the optional `?` is not needed. The implementor isn't required to accommodate the new existence of this object, so it's a backwards-compatible change.

```suggestion
  readonly requestContext: GraphQLRequestContextExecutionDidStart<Record<string, any>>;
```

Reply:
`Record<string, any>` is a fine compromise for now. The `context` typing throughout Apollo Server has some issues that I'd like to tackle, but they are breaking changes and have sweeping effects across many internal and externally facing interfaces and classes. I am happy with this for now.